stdlib/LinearAlgebra/src/matmul.jl

```diff
@@ -142,6 +142,10 @@ function (*)(A::AbstractMatrix, B::AbstractMatrix)
     TS = promote_op(matprod, eltype(A), eltype(B))
     mul!(similar(B, TS, (size(A,1), size(B,2))), A, B)
 end
+function (*)(A::StridedMatrix{<:BlasFloat}, B::StridedMatrix{<:BlasFloat})
+    TS = promote_op(matprod, eltype(A), eltype(B))
```

Comment:
```suggestion
    TS = promote_type(eltype(A), eltype(B))
```

Reply:
Ok, should I change it four lines above as well? This is where I got it from.

Reply:
It might not result in the same type there though (e.g. for general eltypes), but for `BlasFloat` it will so seems better to just use `promote_type` here.